src/paras/Assignment_14/ShopUsingScanner.java

```diff
@@ -0,0 +1,137 @@
+package paras.Assignment_14;
+
+import java.util.Scanner;
+
+import paras.Assignment_7.Shop;
+
+public class ShopUsingScanner {
+	int maggieQuantity; 
+	int dosaQuantity; 
+	int pouchesQuantity; 
+	int panipuriQuantity; 
+	int masalaQuantity;
+	
+	void setInitialQ(int maggieQ, int dosaQ, int pouchesQ, int panipuriQ, int masalaQ){
+		
+		maggieQuantity = maggieQ;
+		dosaQuantity = dosaQ;
+		pouchesQuantity = pouchesQ;
+		panipuriQuantity = panipuriQ;
+		masalaQuantity = masalaQ;
+	}
+	
+	void requestedQuantity(int maggieQ, int dosaQ, int pouchesQ, int panipuriQ, int masalaQ, boolean partialAllowed){
+		
+		if(maggieQuantity < 1){
+			System.out.println("\n Maggie is Out of Stock");
+		}else if(maggieQuantity >= maggieQ){
+			System.out.println("\n As per your request, We will serve Maggie with " + maggieQ + " quantity");
+			maggieQuantity = maggieQuantity - maggieQ;
+		}else if(partialAllowed){
```

Comment:
If you check your line 27th condition, here you are allowed customer to purchase partial items. this flag not required actually, 
Please let me know if you have different understanding for this flag base implementation.